packages/plugins/typescript/typed-document-node/src/index.ts

```diff
@@ -33,22 +33,41 @@ export const plugin: PluginFunction<TypeScriptTypedDocumentNodesConfig> = (
   const visitor = new TypeScriptDocumentNodesVisitor(schema, allFragments, config, documents);
   const visitorResult = oldVisit(allAst, { leave: visitor });
 
+  let content: string[] = [];
+  if (config.documentMode === DocumentMode.string) {
+    content = [
+      `\
+export class TypedDocumentString<TResult, TVariables>
+  extends String
+  implements DocumentTypeDecoration<TResult, TVariables>
+{
+  __apiType?: DocumentTypeDecoration<TResult, TVariables>['__apiType'];
+
+  constructor(private value: string, public __meta__?: { hash: string }) {
+    super(value);
+  }
+
+  toString(): string & DocumentTypeDecoration<TResult, TVariables> {
+    return this.value;
+  }
+}`,
+    ];
+  }
```

Comment:
I'm not exactly sure about this — I didn't want to export it from `@graphql-typed-document-node/core` because there's no need for that IMO — we can inline it in the generated code. But not sure what is the best place in the code to handle that. Having this in `packages/plugins/typescript/typed-document-node/src/index.ts` works, but maybe there's a better approach? Any thoughts, @n1ru4l?

Reply:
yeah I am fine with that place